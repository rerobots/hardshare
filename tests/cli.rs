// SCL <scott@rerobots.net>
// Copyright (C) 2020 rerobots, Inc.

use std::io::prelude::*;

use assert_cmd::Command;
use tempfile::{tempdir, NamedTempFile};


#[test]
fn prints_version() {
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd.arg("-V").assert();
    assert
        .stdout(format!("{}\n", env!("CARGO_PKG_VERSION")))
        .success();
}


#[test]
fn prints_help() {
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd.arg("help").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!(String::from_utf8(output.stdout).unwrap());
}


#[test]
fn config_requires_arg() {
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd.arg("config").assert();
    assert.failure().code(1);
}


#[test]
fn add_token_does_not_exist() {
    let ntf = NamedTempFile::new().unwrap();
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd
        .arg("config")
        .arg("--add-key")
        .arg(ntf.path().join("notexist"))
        .assert();
    assert.failure().code(1);
}


#[test]
fn list_config_does_not_exist() {
    let tmphome = tempdir().unwrap();
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd
        .env("HOME", tmphome.path())
        .arg("config")
        .arg("--local")
        .arg("-l")
        .assert();
    assert.failure().code(1);
}
