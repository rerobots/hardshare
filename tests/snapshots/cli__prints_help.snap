---
source: tests/cli.rs
assertion_line: 26
expression: "String::from_utf8(output.stdout).unwrap()"
---
hardshare 
Command-line interface for the hardshare client

USAGE:
    hardshare [FLAGS] [OPTIONS] [SUBCOMMAND]

FLAGS:
    -h, --help       Prints help information
    -v, --verbose    Increases verboseness level of logs; ignored if RUST_LOG is
                     defined
    -V, --version    Prints version number and exits

OPTIONS:
        --port <PORT>        port for daemon [default: 6666]
        --format <FORMAT>    special output formatting (default is no special
                             formatting); options: YAML , JSON

SUBCOMMANDS:
    ad              Advertise availability, accept new instances
    config          Manage local and remote configuration
    config-addon    Manage add-ons (mistyproxy, vnc, ...)
    declare-org     Declare default organization for commands; for example,
                    `register` will mark the owner as this organization or,
                    if none, the user
    help            Prints this message or the help of the given
                    subcommand(s)
    register        Register new workspace deployment
    rules           Modify access rules (also known as capabilities or
                    permissions)
    stop-ad         Mark as unavailable; optionally wait for current
                    instance to finish
    version         Prints version number and exits

